File: 26-remove-duplicates-from-sorted-array/26-remove-duplicates-from-sorted-array.java
class Solution {
    
    public int removeDuplicates(int[] nums) {
        
        // Solution: O(n) time, O(1) additional space.
        
        if (nums.length < 2) {
            return nums.length;
        }
            
        int i = 0;  // Slow-run pointer.
        
        for (int j = 0; j < nums.length; j++) {
            
            int value = nums[j];
                
            if (value == nums[i]) {
                continue;
            }
            
            // Else, capture the result
            i += 1;
            
            if (i != j) {
                nums[i] = value; // In place overriden.
            }
            
        }
        
        return i + 1;
        
    }
    
}

File: 26-remove-duplicates-from-sorted-array/26-remove-duplicates-from-sorted-array.py
class Solution(object):
    
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        # Solution: O(n) time, O(1) additional space.
        
        if len(nums) < 2:
            return len(nums)
        
        i = 0  # Slow-run pointer.
        for j, value in enumerate(nums):
            if value == nums[i]:
                continue
            # Else, capture the result
            i += 1
            if i != j:
                nums[i] = value # In place overriden.
 
        return i + 1
            
        